src/core/text/TextStyle.js

```diff
@@ -119,7 +124,10 @@ export default class TextStyle
      */
     reset()
     {
-        Object.assign(this, defaultStyle);
+        for (const key in defaultStyle)
+        {
+            deepCopyProperty(this, defaultStyle, key);
+        }
```

Comment:
Why not include the loop inside the deep copy function? Would reduce the couple of loops.

```js
function deepCopy(source, target) {
  for (const prop in defaultStyle) {
    if (Array.isArray(source[prop]) {
       target[prop] = source[prop].slice();
    }
   else {
      target[prop] = source[prop];
   }
  }
}
```

Reply:
Because in the constructor, you want to loop through just the properties passed through in the style object, and in this scenario you want to loop through all properties that are in default style. 

I could add a third parameter which is the object whose properties you want to loop through?

Reply:
That works, and default arg to defaultStyle.

Reply:
Changed made. I didn't make default arg to defaultStyle, because that then causes issues when you pass in nonsense into TextStyle constructor.

For example, give nothing with
```new PIXI.TextStyle()```
And your style object is null, which will kick in the es6 default parameter.

So just explicitly passing through the object to check properties for on the 3 occasions the function is used solved that in the easier way